### ISSUE TITLE
<tts> tag bulk add 

### ISSUE THREAD

**Issue by kabutar111** [AwesomeTTS/AwesomeTTS#82]
how to add <tts> tag art star and end of card, to whole deck automatically on already created cards , i don't want audio to be added to cards , just want anki to read text as Ankidroid does.

**Comment by corpulentcoffee** [comment]
Hello!

You don't have to do this for *every card*, but you just have to do it for *every template* you have (i.e. by clicking on "Cards..." for each note type you have, and going through the tabs).

Do you have many note types?

**Comment by kabutar111** [comment]
Hi! Thank you for your response and AWESOME work on AwesomeTTS.

i have over 20000 cards so quite a few notes type , which are mostly variant of cloze , if you can please in detail explain the step required for same , that would be greatly appreciated. Cut afford to mess up my cards .
once you again thank you so much for your TIME.

**Comment by corpulentcoffee** [comment]
Well, since you're doing a large number of note types, I would recommend that you create a preset before modifying your templates (in case you want to change your TTS service later).

Here's what you do:

1. from the Anki main window, go to **Tools**, **AwesomeTTS**, **Advanced**, **Manage Presets**
2. configure a service how you like it
3. click "Save" and give it an easy name, like `Global` **
4. click OK
5. click OK
6. from the Anki main window, click **Browse**
7. at the bottom of left-hand panel, your note types will be listed; look for an icon like these: 
![screenshot_2015-10-29_09-27-03](https://cloud.githubusercontent.com/assets/3958448/10821224/3df6a5ec-7e1f-11e5-944f-5512478e86a4.png)
8. for each of these note types:
    1. click it; a sample note from that type will come up
    2. in the right-hand panel, click "Cards..."; the template editor will come up
    3. for each tab (i.e. template) you have:
        1. go to the "Front Template" field, and wrap the text you want to be spoken
            in a `<tts>` tag, e.g.: ```<tts preset="Global">{{Front}}</tts>```
        2. go to the "Back Template" field, and wrap the text you want to be spoken in a `<tts>` tag, e.g.:
            ```{{FrontSide}} <hr id=answer> <tts preset="Global">{{Back}}</tts>```
    4. click Close

** If you have more than one TTS service you want to use (e.g. you want to use English on one side and another language on the other side), you can create more than one preset.

**Comment by kabutar111** [comment]
Thanks a lot very very , helpful , and very dumb proof instructions , what you suggest about close deletion cards and {{Extra}} text in it , since they are not showing any {{Back}} , how should i wrap them so that it reads front card and than back side with front and close and extra text there.

Thank you so much 
and i own you a beer if you ever fly to india :)

**Comment by corpulentcoffee** [comment]
Hello!

For the cloze front-sides, you can do something like `<tts preset="Global">{{cloze:Text}}</tts>`.

On the back, you can do either

- `<tts preset="Global">{{cloze:Text}}</tts> {{Extra}}` if you just want the unclozed text read,
- `{{cloze:Text}} <tts preset="Global">{{Extra}}</tts>` if you just want the "extra" field read, or
- `<tts preset="Global">{{Extra}}<br>{{cloze:Text}}</tts>` if you want it *all* read


AwesomeTTS also has some extra options for *how* to read cloze. You can find them on "Text" tab of the configuration:
<img src=https://ankiatts.appspot.com/config.text.png>
This makes it so you can, for example, read only what was deleted on the back side.

**Comment by kabutar111** [comment]
Awesome work , very helpful ,these instruction should be on wiki page also , will help lot of people i think , specially med students

**Comment by kabutar111** [comment]
just a follow up question , is it possible to delete global preset , save it with diff voice and call it global again , will TTS read it normally

**Comment by corpulentcoffee** [comment]
Yes, you can delete it and create a new one by the same name and it will work.

(You can also just save over top of the old one.)

**Comment by kabutar111** [comment]
you mean save next calling it also global , or it will require me to edit the template with new name ?

**Comment by corpulentcoffee** [comment]
If you click "save", you can call it "Global" again, and it will replace the old one.

You will not have to edit your templates.

**Comment by kabutar111** [comment]
just learned that AwesomeTTS saves vice generated in Anki folder , how much size will that take for 20000 cards , considering 15 words sentences in each card. 
Thanks

**Comment by corpulentcoffee** [comment]
It depends mostly on the service, what language you are studying, and whether you're using on-the-fly or stored MP3s.

You can also turn off caching if you want to. See Tools > AwesomeTTS > Advanced > Caching.

<img src=https://ankiatts.appspot.com/config.advanced.png>

Setting this to `0` will clear your entire cache when Anki exits.

**Comment by kabutar111** [comment]
I am using osx built in samantha voice, I am med student, using on the fly tts tag.

I  know about option you described,I actually want TTS to cache, so I can study with voice on my iPad, as that rely on AwesomeTTS cached sound.

Also I go through cards pretty quick and at times AwesomeTTS keeps reading the last card half way through when I already skipped to next card.
Is there anyway to improve on that or its just my Macbook Air is not fast enough.?

Thanks for your continued effort to Help.

**Comment by corpulentcoffee** [comment]
On Mon, Nov 9, 2015 at 5:24 AM, kabutar111 <notifications@github.com> wrote:

> I am using osx built in samantha voice, I am med student, using on the fly
> tts tag.
>
> I know about option you described,I actually want TTS to cache, so I can
> study with voice on my iPad, as that rely on AwesomeTTS cached sound.
>
The "on-the-fly" <tts> tags won't work outside of the desktop client w/
AwesomeTTS installed. For universal access like that, you'll want to store
the MP3s with your collection. If you already have a big collection, the
easiest way to do this is via the card browser -- see
https://ankiatts.appspot.com/usage/browser for


> Also I go through cards pretty quick and at times AwesomeTTS keeps reading
> the last card half way through when I already skipped to next card.
> Is there anyway to improve on that or its just my Macbook Air is not fast
> enough.?
>
This is an Anki feature, actually. Anki will stop playing the audio from
one card if you skip ahead to the next card.
